### PR TITLE
Codespaces devcontainer

### LINKED ISSUE #5904: [CI] Configure Meshery for Codespaces
#### Current Behavior
Later this year, GitHub Codespaces will be made available to the general public. In the meantime, select open source organizations may engage with Codespaces early to both benefit from the tool and offer feedback on its experience and functionality.

#### Desired Behavior
The ability to open Meshery and it's components in Codespaces via a button like this:
image (9)

#### Implementation
The instructions to make Meshery's repos work with Codespaces looks something like this: [customize your Codespace with a development container](https://docs.github.com/en/codespaces/customizing-your-codespace) ([an example](https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/.devcontainer/devcontainer.json)).

#### Acceptance Tests
A single click to get Meshery and its components up and running in Codespaces.

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by sivanaikk**
**Notes for Reviewers**

This PR fixes #5904 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Thanks for this, @sivanaikk 👍

**Comment by acald-creator** [comment]
Reviewing, as I was researching this to see if we could use devcontainers to ease contributor onboarding the projects.

**Comment by rav121** [comment]
There is UI integration test that is failing, us of syntax error in indexui_spec.js.
However, I don't think this changes are related to the tests run.

**Comment by leecalcote** [comment]
Syncing with master to re-run with passing indexui_spec.js.

File: .devcontainer/server/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+# [Choice] Go version (use -bullseye variants on local arm64/Apple Silicon): 1, 1.19, 1.18, 1-bullseye, 1.19-bullseye, 1.18-bullseye, 1-buster, 1.19-buster, 1.18-buster
+ARG VARIANT="1.19-bullseye"
+FROM mcr.microsoft.com/vscode/devcontainers/go:0-${VARIANT}
+
+# [Choice] Node.js version: none, lts/*, 18, 16, 14
+ARG NODE_VERSION="none"
+RUN if [ "${NODE_VERSION}" != "none" ]; then su vscode -c "umask 0002 && . /usr/local/share/nvm/nvm.sh && nvm install ${NODE_VERSION} 2>&1"; fi
+RUN go install golang.org/dl/go1.19.1@latest \
+&& go1.19.1 download
```
Comment (Azanul) [inline .devcontainer/server/Dockerfile:11]: I don't think these two lines (10-11) are necessary as the base image itself comes with both go and its toolchain installed

File: .devcontainer/server/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.245.2/containers/go/.devcontainer/base.Dockerfile
+
+# [Choice] Go version (use -bullseye variants on local arm64/Apple Silicon): 1, 1.19, 1.18, 1-bullseye, 1.19-bullseye, 1.18-bullseye, 1-buster, 1.19-buster, 1.18-buster
+ARG VARIANT="1.19-bullseye"
```
Comment (Azanul) [inline .devcontainer/server/Dockerfile:4]: Default should be "1.19" or "1.19-buster" but if we're correctly mentioning in `devcontainer.json` then it doesn't really matter.

File: .devcontainer/server/Dockerfile (hunk 1 above)
Comment (leecalcote) [reply .devcontainer/server/Dockerfile:11]: Good feedback for @sivanaikk’s consideration.

Comment (leecalcote) [reply .devcontainer/server/Dockerfile:11]: @sivanaikk what do you think?

Comment (sivanaikk) [reply .devcontainer/server/Dockerfile:11]: Previously we were using go1.19.1 in Makefile. Have to update Dockerfile

**Comment by Chadha93** [comment]
@sivanaikk  Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by sivanaikk** [comment]
> @sivanaikk Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

Just added it to the Meeting agenda. Thank you 🙌

File: server/.devcontainer/base.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+# [Optional] Uncomment this section to install additional OS packages.
+# RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+#     && apt-get -y install --no-install-recommends <your-package-list-here>
+
+# [Optional] Uncomment if you want to install an additional version of node using nvm
+# ARG EXTRA_NODE_VERSION=10
+# RUN su node -c "source /usr/local/share/nvm/nvm.sh && nvm install ${EXTRA_NODE_VERSION}"
+
+# [Optional] Uncomment if you want to install more global node modules
+# RUN su node -c "npm install -g <your-package-list-here>""
```
Comment (AugustasV) [inline server/.devcontainer/base.Dockerfile:55]: new line

File: ui/.devcontainer/base.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+# [Optional] Uncomment this section to install additional OS packages.
+# RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+#     && apt-get -y install --no-install-recommends <your-package-list-here>
+
+# [Optional] Uncomment if you want to install an additional version of node using nvm
+# ARG EXTRA_NODE_VERSION=10
+# RUN su node -c "source /usr/local/share/nvm/nvm.sh && nvm install ${EXTRA_NODE_VERSION}"
+
+# [Optional] Uncomment if you want to install more global node modules
+# RUN su node -c "npm install -g <your-package-list-here>""
```
Comment (AugustasV) [inline ui/.devcontainer/base.Dockerfile:55]: new line

**Comment by Chadha93** [comment]
@sivanaikk Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by Azanul** [comment]
@sivanaikk Are you still working on it?

**Comment by Azanul** [comment]
Closing due to inactivity

**Pull request closed without merging**